README.md

```diff
@@ -2,47 +2,65 @@
-| `coverage-reports`                | Optional comma-separated list of reports to send                                       | `''`                                  |
-| `language`                        | Optionally force associating a language with your coverage report                      | `''`                                  |
-| `force-coverage-parser`           | Optionally force using a specific coverage report parser                               | `''`                                  |
+    > ⚠️ **Never write API tokens to your configuration files** and keep your API tokens well protected, as they grant owner permissions to your projects on Codacy.
+
+2.  [Generate a supported code coverage report](https://docs.codacy.com/coverage-reporter/#generating-coverage) on each push to your repository.
+
+3.  Add the following to a file `.github/workflows/codacy-coverage-reporter.yaml` in your repository, where `PATH_TO_REPORT` is the path to the coverage report:
+
+    ```yaml
+    name: Codacy Coverage Reporter
+
+    on: ["push"]
+
+    jobs:
+      codacy-coverage-reporter:
+        runs-on: ubuntu-latest
+        name: codacy-coverage-reporter
+        steps:
+          - uses: actions/checkout@v2
+          - name: Run codacy-coverage-reporter
+            uses: codacy/codacy-coverage-reporter-action@v1
+            with:
+              project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+              # or
+              # api-token: ${{ secrets.CODACY_API_TOKEN }}
+              coverage-reports: <PATH_TO_REPORT>
+              # or a comma-separated list for multiple reports
+              # coverage-reports: <PATH_TO_REPORT>, <PATH_TO_REPORT>
+    ```
+
+4.  Optionally, to add coverage results to GitHub status checks, [configure a quality gate for coverage](https://docs.codacy.com/repositories-configure/integrations/github-integration/#status-checks) and [enable sending status checks](https://docs.codacy.com/repositories-configure/integrations/github-integration/#status-checks) on Codacy.
+
+## Options
+
+The Codacy GitHub Action is a wrapper for running the [Codacy Coverage Reporter CLI](https://github.com/codacy/codacy-coverage-reporter#codacy-coverage-reporter) and supports a subset of the options available for the command `report`:
+
+- `coverage-reports`: path to the coverage report relative to the repository root, or a comma-separated list for multiple reports
+- `project-token`: [project API token](https://docs.codacy.com/codacy-api/api-tokens/#project-api-tokens)
+- `api-token`: [account API token](https://docs.codacy.com/codacy-api/api-tokens/#account-api-tokens), an alternative to `project-token` when setting up multiple repositories
+- `language` (optional): associate a language with your coverage report(s)
+- `force-coverage-parser` (optional): force using a specific coverage report parser
```

Comment:
By listing the options all together, I was forced to make some choices. Let me know if they make sense:

- The title "extra configurations" now should be called "options" or similar, as we're listing all of them.
  - On a related note, these were called "parameters" but I've seen "options" used more often for CLI commands. I'm open to reverting to "parameters".
- I had to reinstate the "optional" flag for `language` and `force-coverage-parser` to differentiate them, since they're not the only ones anymore.
- The last two descriptions are imperative sentences, the first three are sentence fragments (I believe) which makes the list [nonparallel](https://developers.google.com/tech-writing/one/lists-and-tables#exercise_1). Do we care?

Reply:
Now that I'm also having a look at the full list of options, I realize that they're easier to scan and read when formatted in a table like [what we already had before](https://github.com/codacy/codacy-coverage-reporter-action#workflow-options)!

So, my suggestion now is that you keep exactly the same text that you have (which is an improvement over what we currently have live) but use a table to format the information.

> * The title "extra configurations" now should be called "options" or similar, as we're listing all of them.

Makes sense! :+1: 
 
>   * On a related note, these were called "parameters" but I've seen "options" used more often for CLI commands. I'm open to reverting to "parameters".

Also sounds good. :+1: 

> * I had to reinstate the "optional" flag for `language` and `force-coverage-parser` to differentiate them, since they're not the only ones anymore.

Also, notice that the option `coverage-reports` [is also optional](https://github.com/codacy/codacy-coverage-reporter-action/blob/master/action.yml#L14-L17) - if the user doesn't specify the reports, the Codacy Coverage Reporter will try to detect and upload [these supported coverage reports](https://docs.codacy.com/coverage-reporter/#generating-coverage) from the root of the repository.

In fact, all options are optional. But you do need either a project or an account API token for authenticating to Codacy.

> * The last two descriptions are imperative sentences, the first three are sentence fragments (I believe) which makes the list [nonparallel](https://developers.google.com/tech-writing/one/lists-and-tables#exercise_1). Do we care?

It's not a big deal, and you could adjust the last two descriptions to read like this, for example:

"specifies a language to associate with your coverage report"
"forces using a specific coverage report parser"

But the imperative actually seems more straightforward to me.